packages/block-library/src/template-part/edit/use-template-part-post.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+export default function useTemplatePartPost( postId, slug, theme ) {
```

Comment:
The attributes of the template part block can contain postId, slug, and theme e.g:
```<!-- wp:template-part {"postId":28477,"slug":"st1","theme":"twentytwenty"} /-->```
There is some redundancy here, the postId would be enough to retrieve the other two attributes.

Reply:
Template parts that load from files provided by themes can't have post IDs before being customized.

See #18736.

Reply:
Yes that's true I think slug would still exist. What I was saying is when we reference by postId, I think we should not save the slug and theme attributes otherwise we are saving redundant information. For example, the user or a plugin may change the slug of that post or the theme field and in that case, the information stored in attributes referencing that postId would become obsolete.
To reference template parts loaded from files we would still use the slug.

Reply:
>For example, the user or a plugin may change the slug of that post or the theme field and in that case, the information stored in attributes referencing that postId would become obsolete.

That, or deleting the post, would be a way to have instances of the template part go back to rendering the contents from an original theme file.